src/app/features/messages/service/message.service.ts

```diff
@@ -27,6 +28,23 @@ export class MessageService {
     return this.socket.fromEvent(`message:${type}`);
   }
 
+  roomMessage(user: User, type: string): Message {
```

Comment:
Services should be for general purposes in terms of where it should be used, this logic should be moved to the component and removed from here